models/evc.py

```diff
@@ -946,7 +951,12 @@ def deploy_to_path(self, path=None, old_path_dict: dict = None):
                 use_path = Path()
                 self._install_direct_uni_flows()
             else:
-                msg = f"{self} was not deployed. No available path was found."
+                no_path_msg = "No available path was found."
+                if no_valid_path:
```

Comment:
That's a good augmentation to have in the logs considering the state of application error handling that we have. 

Can you also try to assert for the "try increasing `max_paths`" string in a unit test in a mocked log?

Reply:
Added unit test.